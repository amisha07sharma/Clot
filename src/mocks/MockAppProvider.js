import React, { useReducer } from "react";
import PropTypes from "prop-types";
import { AppContext } from "../context/AppContext";
import appReducer from "../reducers/appReducer";

const initialState = {
  email: "mocked_email@example.com",
  firstName: "John",
  lastName: "Doe",
};

const MockAppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(appReducer, initialState);

  return (
    <AppContext.Provider value={{ state, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

MockAppProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MockAppProvider;
