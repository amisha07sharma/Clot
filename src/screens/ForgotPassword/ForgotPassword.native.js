import React, { useState, useContext } from "react";
import PropTypes from "prop-types";
import useValidation from "../../hooks/useValidation";
import { AppContext } from "../../context/AppContext";
import CLOT from "../../constants/messages.json";
import {
  MainContainer,
  HeadingText,
  TextInputContainer,
} from "../../styles/common.style";
import CustomButton from "../../components/CustomButton";

const ForgotPassword = ({ onContinue }) => {
  const { state, dispatch } = useContext(AppContext);
  const { emailValidation } = useValidation();
  const [email, setEmail] = useState(null);
  const handleEmailChange = (text) => {
    setEmail(text);
  };
  const onContinuePress = () => {
    dispatch({ type: "UPDATE_EMAIL", payload: email });
    onContinue();
  };

  return (
    <MainContainer>
      <HeadingText>{CLOT.FORGOT_PASSWORD_SCREEN.FORGOT_PASSWORD}</HeadingText>
      <TextInputContainer
        testID="EmailInput"
        onChangeText={handleEmailChange}
        placeholder={CLOT.FORGOT_PASSWORD_SCREEN.ENTER_EMAIL}
        value={email}
      />
      <CustomButton
        testID="ContinueButton"
        title={CLOT.FORGOT_PASSWORD_SCREEN.CONTINUE}
        onPress={onContinuePress}
        disabled={!emailValidation(email)}
      />
    </MainContainer>
  );
};

ForgotPassword.propTypes = {
  onContinue: PropTypes.func.isRequired,
};

export default ForgotPassword;
