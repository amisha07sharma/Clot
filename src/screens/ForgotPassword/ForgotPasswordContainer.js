import React, { useEffect } from "react";
import PropTypes from "prop-types";
import ForgotPassword from "./ForgotPassword";

const ForgotPasswordContainer = ({ navigation }) => {
  const onContinue = () => {
    navigation.navigate("ResetPassword");
  };
  return <ForgotPassword onContinue={onContinue} testID="ForgotPassword" />;
};

ForgotPasswordContainer.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default ForgotPasswordContainer;
