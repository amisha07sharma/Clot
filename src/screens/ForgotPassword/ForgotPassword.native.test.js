import React from "react";
import { View as MockView } from "react-native";
import { render, fireEvent, screen } from "@testing-library/react-native";
import MockAppProvider from "../../mocks/MockAppProvider";
import ForgotPassword from "./ForgotPassword";

jest.mock("../../hooks", () => ({
  useValidation: () => jest.fn(),
}));

jest.mock("../../components/CustomButton", () => (props) => (
  <MockView {...props} />
));

const mockOnContinue = jest.fn();
const mockProps = {
  onContinue: mockOnContinue,
};

const renderComponent = (props = {}) => {
  return render(
    <MockAppProvider>
      <ForgotPassword {...mockProps} {...props} />
    </MockAppProvider>
  );
};
describe("Tests for ForgotPassword Screen", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match the snapshot", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("updates email state when text input changes", () => {
    renderComponent();

    const emailInput = screen.getByTestId("EmailInput");

    fireEvent.changeText(emailInput, "new_email@example.com");
    expect(emailInput.props.value).toBe("new_email@example.com");
  });

  it("calls onContinue function when Continue button is pressed", () => {
    renderComponent();

    fireEvent(screen.getByTestId("ContinueButton"), "onPress");

    expect(mockOnContinue).toHaveBeenCalledTimes(1);
  });
});
