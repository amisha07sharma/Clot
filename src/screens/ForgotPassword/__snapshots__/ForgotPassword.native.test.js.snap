// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ForgotPassword Screen should match the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "height": "100%",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 50,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 32,
        "fontWeight": "700",
        "letterSpacing": -0.41,
        "lineHeight": 34.5,
        "marginBottom": 30,
      }
    }
  >
    Forgot Password
  </Text>
  <TextInput
    backgroundColor="#f4f4f4"
    borderRadius={4}
    borderWidth={0.1}
    fontSize={16}
    height={56}
    letterSpacing={-0.41}
    lineHeight={20.24}
    marginBottom={20}
    onChangeText={[Function]}
    paddingLeft={10}
    placeholder="Enter Email Address"
    style={
      {
        "fontWeight": "450px",
      }
    }
    testID="EmailInput"
    theme={{}}
    value={null}
  />
  <View
    disabled={true}
    onPress={[Function]}
    testID="ContinueButton"
    title="Continue"
  />
</View>
`;
